packages/ui/src/components/vuestic-components/va-datepicker/components/VaDayPicker/VaDatePickerCalendarHook.ts

```diff
@@ -0,0 +1,65 @@
+import { computed, Ref, ref } from 'vue'
+
+/** Returns last day of previous month */
+export const getMonthDaysCount = (year: number, month: number): number => new Date(year, month + 1, 0).getDate()
+
+export const getMonthStartWeekday = (year: number, month: number) => new Date(year, month, 1).getDay()
+
+/** Returns array from 1 to length */
+export const getNumbersArray = (length: number) => Array.from(Array(length).keys()).map((k) => k + 1)
+
+export const useVaDatePickerCalendar = (currentYear: Ref<number>, currentMonth: Ref<number>, options?: { firstWeekday?: Ref<'Monday' | 'Sunday'> }) => {
+  const CALENDAR_ROWS_COUNT = 6 // Need 6 rows if first day of a month is Saturday and the last day is Monday 31th.
+
+  const localizeWeekday = (weekdayNumber: number) => {
+    if (options && options.firstWeekday?.value === 'Monday') {
+      // Set Sunday as 7th day of the week and Monday as first day of the week.
+      return weekdayNumber === 0 ? 6 : weekdayNumber - 1
+    }
+
+    return weekdayNumber
+  }
+
+  const getPreviousDates = () => {
```

Comment:
also pretty sure we want to use date-fns here.

Reply:
I can see any date-fns helper that can be useful here.